lib/client.js

```diff
@@ -588,7 +588,12 @@ MqttClient.prototype.end = function (force, cb) {
   function closeStores () {
     that.disconnected = true
     that.incomingStore.close(function () {
-      that.outgoingStore.close(cb)
+      that.outgoingStore.close(function () {
```

Comment:
this can pass an error as well.

Reply:
You are right, I have blindly only considered initially the default store implementation but I have also to consider any external implementation.
I will pass any argument as is (to the `end` event as well).

Reply:
no error to `'end'` please.

Reply:
Done with no error on `end` event :-).
I have also added a test to check that the error is passed up to the `end()` callback.